tests/storage/hotplug.go

```diff
@@ -1265,9 +1272,92 @@ var _ = SIGDescribe("Hotplug", func() {
+				Skip("Skip tests when CDI is not present")
+			}
+			var exists bool
+			sc, exists = libstorage.GetRWXBlockStorageClass()
+			if !exists {
+				Skip("Skip test when RWXBlock storage class is not present")
+			}
+		})
+
+		AfterEach(func() {
+			if vm != nil {
+				err := virtClient.VirtualMachine(vm.Namespace).Delete(vm.Name, &metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				vm = nil
+			}
+
+			if dv != nil {
+				err = virtClient.CdiClient().CdiV1beta1().DataVolumes(dv.Namespace).Delete(context.Background(), dv.Name, metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				dv = nil
+			}
+		})
+
+		DescribeTable("should be able to add and remove volumes with disk mutating hook sidecars", func(addVolumeFunc addVolumeFunction, removeVolumeFunc removeVolumeFunction, volumeMode corev1.PersistentVolumeMode, vmiOnly bool) {
+			var err error
+			url := "docker://" + cd.ContainerDiskFor(cd.ContainerDiskCirros)
+			dv = libstorage.NewDataVolumeWithRegistryImport(url, util.NamespaceTestDefault, k8sv1.ReadWriteOnce)
+			tests.SetDataVolumeForceBindAnnotation(dv)
+			dv.Spec.PVC.Resources.Requests["storage"] = resource.MustParse("5Gi")
+			dv, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(dv.Namespace).Create(context.Background(), dv, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("waiting for the dv import to pvc to finish")
+			libstorage.EventuallyDV(dv, 180, HaveSucceeded())
+
+			By("rename disk image on PVC")
+			pvc, err := virtClient.CoreV1().PersistentVolumeClaims(dv.Namespace).Get(context.Background(), dv.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			tests.RenameImgFile(pvc, newDiskImgName)
+
+			By("start VM with disk mutation sidecar")
+			vmi := tests.NewRandomVMIWithDataVolume(dv.Name)
+			vmi.ObjectMeta.Annotations = renderSidecar()
+			vm := tests.NewRandomVirtualMachine(vmi, true)
+			vm, err = virtClient.VirtualMachine(util.NamespaceTestDefault).Create(vm)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() bool {
+				vm, err := virtClient.VirtualMachine(util.NamespaceTestDefault).Get(vm.Name, &metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return vm.Status.Ready
+			}, 300*time.Second, 1*time.Second).Should(BeTrue())
+
+			verifyAttachDetachVolume(vm, addVolumeFunc, removeVolumeFunc, sc, volumeMode, vmiOnly, true)
+		},
+			Entry("with DataVolume and running VM", addDVVolumeVM, removeVolumeVM, corev1.PersistentVolumeFilesystem, false),
+			Entry("with Block DataVolume immediate attach", addDVVolumeVM, removeVolumeVM, corev1.PersistentVolumeBlock, false),
+			Entry("with DataVolume and VMI directly", addDVVolumeVMI, removeVolumeVMI, corev1.PersistentVolumeFilesystem, true),
+		)
+	})
 })
 
-func VerifyVolumeAndDiskVMAdded(virtClient kubecli.KubevirtClient, vm *v1.VirtualMachine, volumeNames ...string) {
+func verifyVolumeAndDiskVMAdded(virtClient kubecli.KubevirtClient, vm *v1.VirtualMachine, volumeNames ...string) {
```

Comment:
Why are you making this function private?

Reply:
I wasn't sure why it was public when we use it only in the storage package. I can revert the change if it was left public for a reason.

Reply:
It was likely used in the memory dump tests, which might not have been in the storage package originally. I think it is fine private.

Reply:
This function was previously in `tests/utils.go` where it had to be public. I guess it is moved here when the test packages were restructured. I don't see it being used anywhere, so I assume it is safe to keep it private.